lib/trino-filesystem-manager/src/main/java/io/trino/filesystem/manager/FileSystemModule.java

```diff
@@ -104,9 +108,14 @@ protected void setup(Binder binder)
         newOptionalBinder(binder, CacheKeyProvider.class).setDefault().to(DefaultCacheKeyProvider.class).in(Scopes.SINGLETON);
 
         newOptionalBinder(binder, TrinoFileSystemCache.class);
+        newOptionalBinder(binder, MemoryFileSystemCache.class);
 
+        boolean isCoordinator = nodeManager.getCurrentNode().isCoordinator();
         if (config.isCacheEnabled()) {
-            install(new AlluxioFileSystemCacheModule(nodeManager.getCurrentNode().isCoordinator()));
+            install(new AlluxioFileSystemCacheModule(isCoordinator));
+        }
+        if (enableCoordinatorFilesCache) {
+            install(new MemoryFileSystemCacheModule(isCoordinator));
```

Comment:
`isCoordinator` could be `false` here

Reply:
see https://github.com/trinodb/trino/pull/22739/commits/d0ec97ee983da24d71935f9ea23f56952a8e4128#r1696886201